lib/ruby-mpd/song.rb

```diff
@@ -24,6 +24,18 @@ def ==(another)
     self.class == another.class && self.file == another.file
   end
 
+  def to_json(*a)
+    {
+      "time" => @time,
+      "file" => @file,
+      "title" => @title,
+      "artist" => @artist,
+      "album" => @album,
+      "albumartist" => @albumartist,
+      "data" => @data
+    }.to_json(*a)
```

Comment:
Also, `to_json` is a rails construct, and as such, this method call will fail on vanilla ruby. Instead, refactor this method into `to_h`, which should return a hash. Then you can call `to_json`/`as_json` on it as needed inside your application (or do any other form of serialization).